custom_model_runner/datarobot_drum/drum/main.py

```diff
@@ -29,38 +29,52 @@
     # Run regression user model in fit mode.
     drum fit --code-dir <custom code dir> --input <input.csv> --output <output_dir> --target <target feature> --verbose
 """
-
-import argcomplete
 import os
 import sys
-import signal
-import requests
 from datarobot_drum.drum.args_parser import CMRunnerArgsRegistry
-from datarobot_drum.drum.drum import CMRunner
+from datarobot_drum.drum.runtime import DrumRuntime
 
 
 def main():
-    arg_parser = CMRunnerArgsRegistry.get_arg_parser()
-    # argcomplete call should be as close to the beginning as possible
-    argcomplete.autocomplete(arg_parser)
-    options = arg_parser.parse_args()
-    CMRunnerArgsRegistry.verify_options(options)
-
-    def signal_handler(sig, frame):
-        # The signal is assigned so the stacktrace is not presented when Ctrl-C is pressed.
-        # The cleanup itself is done only if we are NOT running in performance test mode which
-        # has its own cleanup
-        print("Ctrl+C pressed, aborting cmrun")
-
-        if options.docker and not options.in_perf_mode_internal:
-            url = "http://{}/shutdown/".format(options.address)
-            print("Sending shutdown to server: {}".format(url))
-            requests.post(url, timeout=2)
-        os.system("tput init")
-        sys.exit(0)
-
-    signal.signal(signal.SIGINT, signal_handler)
-    CMRunner(options).run()
+    with DrumRuntime() as runtime:
+        arg_parser = CMRunnerArgsRegistry.get_arg_parser()
+
+        try:
+            import argcomplete
+        except ImportError:
+            print(
+                "WARNING: autocompletion of arguments is not supported "
+                "as 'argcomplete' package is not found",
+                file=sys.stderr,
+            )
+        else:
+            # argcomplete call should be as close to the beginning as possible
+            argcomplete.autocomplete(arg_parser)
+
+        options = arg_parser.parse_args()
+        CMRunnerArgsRegistry.verify_options(options)
+
+        runtime.options = options
+
+        import signal
+        import requests
+        from datarobot_drum.drum.drum import CMRunner
+
+        def signal_handler(sig, frame):
+            # The signal is assigned so the stacktrace is not presented when Ctrl-C is pressed.
+            # The cleanup itself is done only if we are NOT running in performance test mode which
+            # has its own cleanup
+            print("Ctrl+C pressed, aborting cmrun")
+
+            if options.docker and not options.in_perf_mode_internal:
+                url = "http://{}/shutdown/".format(options.address)
+                print("Sending shutdown to server: {}".format(url))
+                requests.post(url, timeout=2)
+            os.system("tput init")
+            sys.exit(0)
```

Comment:
@4dahalibut changed this place, rebase on master

Reply:
done